libs/trainer/mlpe/trainer/optimizers.py

```diff
@@ -0,0 +1,34 @@
+import inspect
+from typing import Iterable
+
+import torch
+
+
+def _wrap_optimizer(optimizer):
```

Comment:
Can we try to get this double-level wrapper inside __init__.py, and call it something like `_core_wrapper`, and reuse it for both optimizer and scheduler. In one case, the first arg is the model parameters, in the other it is the optimizer, but the wrapper is agnostic to that (at least looks like so)

Reply:
Yup thats a good point. Let me try to generalize it.